lib/spack/llnl/util/tty/log.py

```diff
@@ -38,32 +39,40 @@
 
 
 @contextmanager
-def background_safe():
-    signal.signal(signal.SIGTTOU, signal.SIG_IGN)
+def ignore_signal(signum):
+    """Context manager to temporarily ignore a signal."""
+    old_handler = signal.getsignal(signum)
+    signal.signal(signum, signal.SIG_IGN)
     yield
-    signal.signal(signal.SIGTTOU, signal.SIG_DFL)
+    signal.signal(signum, old_handler)
 
 
-def _is_background_tty():
-    """Return True iff this process is backgrounded and stdout is a tty"""
-    if sys.stdout.isatty():
-        return os.getpgrp() != os.tcgetpgrp(sys.stdout.fileno())
-    return False  # not writing to tty, not background
+def _is_background_tty(stream):
```

Comment:
> This is not correct because of the check for stream.isatty()

I'm not sure how it's not correct.  `os.getpgrp()` gets my process's group id.  It's checking whether or not it's in the foreground for the terminal attached to `stream`.

Reply:
When you say "IFF" do you mean "if and only if"?

Reply:
yes

Reply:
Ah so what I mean when I say this is wrong is that this function can return `False` if `stream.isatty() == False`, but it is possible that there are non-tty FG streams.

So I think the following wording is better:

> Return True if the stream is a tty and is in the background.

In that wording, you could also replace "if" with "iff", but I think it may just be distracting.

Reply:
I kind of don't like that wording because "the stream" is never in the background.  It's *the calling process* that is either fg or bg.  Do you have a suggestion for how to work that in?

Reply:
> Return True if the stream is a tty and the calling process is in the background

Reply:
done.